aten/src/ATen/native/cuda/RowwiseScaledMM.cu

```diff
@@ -281,6 +283,249 @@ void f8f8bf16_rowwise_impl(
   C10_CUDA_KERNEL_LAUNCH_CHECK();
 }
 
+// Cutlass rowwise kernel for SM89
+template <
+    typename FastAccum,
+    typename DtypeA,
+    typename DtypeB,
+    typename DtypeBias>
+void f8f8bf16_rowwise_impl_sm89(
+    at::Tensor XQ, // FP8
+    at::Tensor WQ, // FP8
+    at::Tensor x_scale,
+    at::Tensor w_scale,
+    std::optional<at::Tensor> bias,
+    at::Tensor out) {
+  int M = XQ.size(0);
+  int N = WQ.size(1);
+  int K = XQ.size(1);
+
+  using LayoutInputA = cutlass::layout::RowMajor;
+  constexpr int AlignmentInputA = 16 / sizeof(DtypeA);
+
+  using LayoutInputB = cutlass::layout::ColumnMajor;
+  constexpr int AlignmentInputB = 16 / sizeof(DtypeB);
+
+  constexpr int AlignmentScale = 16 / sizeof(DtypeScale);
+  constexpr int AlignmentBias = 16 / sizeof(DtypeBias);
+
+  using LayoutOutput = cutlass::layout::RowMajor;
+  constexpr int AlignmentOutput = 16 / sizeof(DtypeOutput);
+
+  // Tag indicating the minimum SM that supports the intended feature
+  using ArchTag = cutlass::arch::Sm89;
+  using OperatorClass = cutlass::arch::OpClassTensorOp;
+
+  using ThreadblockSwizzle =
+      cutlass::gemm::threadblock::ThreadblockSwizzleStreamK;
+
+  // TODO: instead of fixing these values, implement logic alike to
+  // what is used for SM90+.
+  using ThreadblockShape = cutlass::gemm::GemmShape<128, 128, 64>;
```

Comment:
agree w/ comment I think sm89 has have the amount of shared memory and this might spill espeically for pingpong w/ slow-accum